src/core/Client.ts

```diff
@@ -132,18 +133,35 @@ export class Client<TABITable extends ABITable> {
     }))[0]!;
   }
 
+  /**
+   * Builds ABI from a provided address and module name for given client. ABI name can be taken from abi.name
+   * 
+   * @param address The module address
+   * @param moduleName The module name
+   * @returns The constructed ABI
+   * @example
+   * const abi = await client.fetchABI(address = '0x1', moduleName = 'AptosCoin');
+   */
+  public async fetchABI<T extends ABIRoot>(address: string, moduleName: string): Promise<T> {
+      // Fetches ABI fom address and module name for given client
+      // throws if inexistent module name in address for given client
+      return (await this.client.getAccountModule({ accountAddress: address, moduleName: moduleName })).abi as unknown as T;
+  }
+
   /**
    * Create a client associated with a specific ABI.
    *
    * @param abi The ABI JSON.
+   * @param address The address of the module. If not provided, ABI address will be used.
    * @returns A client can call view/entry functions or get account resource.
    * @example
    * const [balance] = await client.useABI(COIN_ABI).view.balance({
    *    functionArguments: ['0x1'],
    *    typeArguments: ['0x1::aptos_coin::AptosCoin'],
    * });
    */
-  public useABI<T extends ABIRoot>(abi: T) {
+  public useABI<T extends ABIRoot>(abi: T, address?: string) {
+    if (address) abi.address = address;
```

Comment:
It's better not to directly modify ﻿abi since this would alter the ﻿abi object that the user passed in without their knowledge.

For example, if the user's ﻿`abi` object has `﻿abi.address` as ﻿0x123, and after passing it in, `﻿abi.address` changes to ﻿0x456, users might notice unexpected address changes when using the ﻿abi object elsewhere.

I think it's better to add a `overrideAddress` local variable here. And add another optional  `address` in the `createViewPayload` below.

For example:
```
createViewPayload(abi, {
          function: functionName,
          typeArguments: args[0].typeArguments,
          functionArguments: args[0].functionArguments,
          address: overrideAddress,
});
```